src/common/Finisher.cc

```diff
@@ -3,6 +3,11 @@
 
 #include "Finisher.h"
 
+#include "common/debug.h"
+
+// re-include our assert to clobber the system one; fix dout:
```

Comment:
@tchaikov I've removed references to "include/atomic.h", and I'd imagine that's the real cause, though unfortunately it's a bit hard for me to cleanly say if that's the "real reason"(TM), since early on I didn't have this correctly based of off master. More than happy if there are alternatives, but I'm not sure what those would be-- certainly possible that I'm doing something else wrong that's causing the issue, but I don't know what it would be.